extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultSystemBackendManager.java

```diff
@@ -0,0 +1,32 @@
+package io.quarkus.vault.runtime;
+
+import io.quarkus.vault.VaultSystemBackendEngine;
+import io.quarkus.vault.runtime.client.VaultClient;
+import io.quarkus.vault.runtime.config.VaultRuntimeConfig;
+import io.quarkus.vault.runtime.sys.VaultHealth;
+
+public class VaultSystemBackendManager implements VaultSystemBackendEngine {
+
+    private VaultClient vaultClient;
+    private VaultRuntimeConfig serverConfig;
+
+    public VaultSystemBackendManager(VaultClient vaultClient, VaultRuntimeConfig serverConfig) {
+        this.vaultClient = vaultClient;
+        this.serverConfig = serverConfig;
+    }
+
+    @Override
+    public VaultHealth health(boolean isStandByOk, boolean isPerfStandByOk) {
```

Comment:
Would you like to keep these parameters to let the user code override the configuration properties ? I propose just drop them since the config properties win anyway and the user code does not probably need to know about these minor status tweaks

Reply:
I copy the same comment before just to keep a relationship:

I know that you are asking why are you leaving this instead of provide a health() method. The reason is that VaultSystemBackendEngine could be injected in any class and users programmatically might want to set them programmatically.